gcloud-java-compute/src/main/java/com/google/gcloud/compute/InstanceInfo.java

```diff
@@ -0,0 +1,1282 @@
+     */
+    public static Metadata of(Map<String, String> values) {
+      return builder().values(values).build();
+    }
+
+    static Metadata fromPb(com.google.api.services.compute.model.Metadata metadataPb) {
+      Builder builder = builder();
+      if (metadataPb.getItems() != null) {
+        Map<String, String> metadataValues =
+            Maps.newHashMapWithExpectedSize(metadataPb.getItems().size());
+        for (com.google.api.services.compute.model.Metadata.Items item : metadataPb.getItems()) {
+          metadataValues.put(item.getKey(), item.getValue());
+        }
+        builder.values(metadataValues);
+      }
+      return builder.fingerprint(metadataPb.getFingerprint()).build();
+    }
+  }
+
+  /**
+   * A service account, with its specified scopes, authorized for this instance.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/authentication">Authenticating from Google
+   *     Compute Engine</a>
+   */
+  public static final class ServiceAccount implements Serializable {
+
+    static final Function<com.google.api.services.compute.model.ServiceAccount, ServiceAccount>
+        FROM_PB_FUNCTION =
+        new Function<com.google.api.services.compute.model.ServiceAccount, ServiceAccount>() {
+          @Override
+          public ServiceAccount apply(com.google.api.services.compute.model.ServiceAccount pb) {
+            return ServiceAccount.fromPb(pb);
+          }
+        };
+    static final Function<ServiceAccount, com.google.api.services.compute.model.ServiceAccount>
+        TO_PB_FUNCTION =
+        new Function<ServiceAccount, com.google.api.services.compute.model.ServiceAccount>() {
+          @Override
+          public com.google.api.services.compute.model.ServiceAccount apply(
+              ServiceAccount metadata) {
+            return metadata.toPb();
+          }
+        };
+
+    private static final long serialVersionUID = 4199610694227857331L;
+
+    private final String email;
+    private final List<String> scopes;
+
+    private ServiceAccount(String email, List<String> scopes) {
+      this.email = email;
+      this.scopes = ImmutableList.copyOf(scopes);
+    }
+
+    /**
+     * Returns the email address of the service account.
+     */
+    public String email() {
+      return email;
+    }
+
+    /**
+     * Returns the list of scopes to be made available for this service account.
+     */
+    public List<String> scopes() {
+      return scopes;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("email", email)
+          .add("scopes", scopes)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(email, scopes);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj == this
+          || obj instanceof ServiceAccount
+          && Objects.equals(toPb(), ((ServiceAccount) obj).toPb());
```

Comment:
This seems pretty common -- any way to abstract up? If so, maybe file an issue to do it? (rather than hold up a merge)